nservicebus/messaging/send-a-message.md

```diff
@@ -90,10 +90,23 @@ snippet: BasicSendReplyToDestination
 
 While it's usually best to let NServiceBus [handle all exceptions](/nservicebus/recoverability/), there are some scenarios where messages might need to be sent regardless of whether the message handler succeeds or not, for example, to send a reply notifying that there was a problem with processing the message.
 
-NOTE: Side effects can occur when failures happen after sending the message. The messages could be retried meaning duplicate messages are created if this code is executed more than once. When messages are sent via immediate disaptch, ensure that the same [message identifier](/nservicebus/messaging/message-identity.md) gets assigned to them when invoked more than once. Second, due to failures it could happen that messages are sent that contain state which is inconsistent because of failing operations like a storage modification that didn't occur.
+### Usage
 
 This can be done by using the immediate dispatch API:
 
 snippet: RequestImmediateDispatch
 
-WARNING: By specifying immediate dispatch, outgoing messages will not be [batched](/nservicebus/messaging/batched-dispatch.md) or enlisted in the current receive transaction, even if the transport supports transactions or batching. Similarly, when the [outbox](/nservicebus/outbox/) feature is enabled, messages sent using immediate dispatch won't go through the outbox.
+NOTE: The API does not impact sending outside a incoming message context like when invoking message operations on `IMessageSession` for example send-only endpoints.
```

Comment:
Should we mention it would impact sends done with the transactional session or is that clear enough with the section talking about the outbox?

Reply:
```suggestion
NOTE: The API behaves the same for `ITransactionalSession` but differently for `IMessageSession` when sending outside an incoming message context. For example when invoking message operations on `IMessageSession` the `RequestImmediateDispatch` does not have any effect.
```